File: h2o-hadoop-common/tests_fault_tolerance/R/test-frame-reload.R
source("fault_tolerance_utils.R")

name_node <- Sys.getenv("NAME_NODE")
work_dir <- sprintf("hdfs://%s%s", name_node, get_workdir())
dataset <- "/datasets/mnist/train.csv.gz"

test_that("Can reload a Frame from hdfs", {
    print("Import and save...")
    saver_cluster_name <- "saver-r"
    tryCatch({
        cluster_1 <- start_cluster(saver_cluster_name)
        h2o.connect(ip=cluster_1[1], port=as.numeric(cluster_1[2]))
        df_orig <- h2o.importFile(path=sprintf("hdfs://%s%s", name_node, dataset))
        df_key <- h2o.getId(df_orig)
        df_orig_r <- as.data.frame(df_orig)
        h2o.save_frame(df_orig, work_dir)
    }, finally={
        stop_cluster(saver_cluster_name)
    })

    print("Load saved...")
    loader_cluster_name <- "loader-r"
    tryCatch({
        cluster_2 <- start_cluster(loader_cluster_name)
        h2o.connect(ip=cluster_2[1], port=as.numeric(cluster_2[2]))
        df_loaded <- h2o.load_frame(df_key, work_dir)
        df_loaded_r <- as.data.frame(df_loaded)
    }, finally={
        stop_cluster(loader_cluster_name)
    })

    print("Comparing frames...")
    # just a basic comparison, more testing done in py
    expect_true(all(dim(df_orig_r) == dim(df_loaded_r)))

    print("Test passed.")
})

File: h2o-hadoop-common/tests_fault_tolerance/python/test_frame_reload.py
from __future__ import print_function
import sys
import os
sys.path.insert(1, os.path.join("..", "..", "..", "h2o-py"))
from tests import pyunit_utils
import fault_tolerance_utils as utils
import h2o
import unittest


class FrameReloadTest(unittest.TestCase):

    def test_frame_reload(self):
        name_node = pyunit_utils.hadoop_namenode()
        work_dir = "hdfs://%s%s" % (name_node, utils.get_workdir())
        dataset = "/datasets/mnist/train.csv.gz"
        
        saver_cluster_name = "saver-py" 
        try:
            cluster_1 = utils.start_cluster(saver_cluster_name)
            h2o.connect(url=cluster_1)
            df_orig = h2o.import_file(path="hdfs://%s%s" % (name_node, dataset))
            df_key = df_orig.key
            df_pd_orig = df_orig.as_data_frame()
            df_orig.save(work_dir)
            h2o.connection().close()
        finally:
            utils.stop_cluster(saver_cluster_name)
        
        loader_cluster_name = "loader-py"
        try:
            cluster_2 = utils.start_cluster(loader_cluster_name)
            h2o.connect(url=cluster_2)
            df_loaded = h2o.load_frame(df_key, work_dir)
            df_pd_loaded = df_loaded.as_data_frame()
            h2o.connection().close()
        finally:
            utils.stop_cluster(loader_cluster_name)
        
        self.assertTrue(df_pd_orig.equals(df_pd_loaded))


if __name__ == '__main__':
    unittest.main()
